services/httpd/handler_test.go

```diff
@@ -698,6 +698,60 @@ func TestHandler_XForwardedFor(t *testing.T) {
 	}
 }
 
+func TestHandler_XRequestID(t *testing.T) {
+	var buf bytes.Buffer
+	h := NewHandler(false)
+	h.CLFLogger = log.New(&buf, "", 0)
+
+	cases := []struct {
+		XRequestID string
+		RequestId  string
+	}{
+		{XRequestID: "abc123", RequestId: ""},          // X-Request-ID is used.
+		{XRequestID: "", RequestId: "foobarzoo"},       // Request-Id is used.
+		{XRequestID: "abc123", RequestId: "foobarzoo"}, // X-Request-ID takes precedence.
+		{XRequestID: "", RequestId: ""},                // v1 UUID generated.
+	}
+
+	for _, c := range cases {
+		t.Run(fmt.Sprintf("X-Request-ID: %q Request-Id: %q", c.XRequestID, c.RequestId), func(t *testing.T) {
+			buf.Reset() // Reset the buffer.
```

Comment:
![](https://media1.popsugar-assets.com/files/thumbor/QUNrEBTrKGluzJpYq17uRNlEwQI/fit-in/1024x1024/filters:format_auto-!!-:strip_icc-!!-/2015/03/15/202/n/1922283/dcd33b79_I9sR3CR/i/Animal-House.gif)